tests/airmar_tests/test_nmeaparser.py

```diff
@@ -41,3 +41,143 @@ def test_settings(self):
+        self.assertEqual(data["GPGGA"]["hdop"], "0.9")
+        self.assertEqual(data["GPGGA"]["geoid"], "545.4")
+        self.assertEqual(data["GPGGA"]["geiodal_separation"], "46.9")
+        self.assertEqual(data["GPGGA"]["age_of_differential_gps_data"], None)
+        self.assertEqual(data["GPGGA"]["differential_reference_station_id"], None)
+
+    def test_update_sentence_wimwd(self):
+        data = {}
+        sentence = "WIMWD,359.9,T,359.9,M,10.1,N,19.1,M"
+        sentence = self._make_nmea_sentence(sentence=sentence)
+
+        fields = self.parser.parse(sentence=sentence)
+        self.parser.update_data(data=data, fields=fields)
+
+        self.assertEqual(data["WIMWD"]["wind_direction_true"], "359.9")
+        self.assertEqual(data["WIMWD"]["wind_direction_magnetic"], "359.9")
+        self.assertEqual(data["WIMWD"]["wind_speed_knots"], "10.1")
+        self.assertEqual(data["WIMWD"]["wind_speed_mps"], "19.1")
+
+    def test_update_sentence_gpvtg(self):
+        data = {}
+        sentence = "GPVTG,054.7,T,034.4,M,005.5,N,010.2,K,A"
+        sentence = self._make_nmea_sentence(sentence=sentence)
+
+        fields = self.parser.parse(sentence=sentence)
+        self.parser.update_data(data=data, fields=fields)
+
+        self.assertEqual(data["GPVTG"]["course_over_ground_true"], "054.7")
+        self.assertEqual(data["GPVTG"]["course_over_ground_magnetic"], "034.4")
+        self.assertEqual(data["GPVTG"]["speed_over_ground_knots"], "005.5")
+        self.assertEqual(data["GPVTG"]["speed_over_ground_kph"], "010.2")
+        self.assertEqual(data["GPVTG"]["mode_indicator"], "A")
+
+    def test_update_sentence_wivwr(self):
+        data = {}
+        sentence = "WIVWR,010.1,L,020.2,N,10.1,M,10.2,K"
+        sentence = self._make_nmea_sentence(sentence=sentence)
+        
+        fields = self.parser.parse(sentence=sentence)
+        self.parser.update_data(data=data, fields=fields)
+
+        self.assertEqual(data["WIVWR"]["wind_angle_degree"], "010.1")
+        self.assertEqual(data["WIVWR"]["wind_angle_direction"], "L")
+        self.assertEqual(data["WIVWR"]["wind_speed_knots"], "020.2")
+        self.assertEqual(data["WIVWR"]["wind_speed_mps"], "10.1")
+        self.assertEqual(data["WIVWR"]["wind_speed_kph"], "10.2")
+
+    def test_update_sentence_wivwt(self):
+        data = {}
+        sentence = "WIVWT,0,L,1,N,2,M,3,K"
+        sentence = self._make_nmea_sentence(sentence=sentence)
+
+        fields = self.parser.parse(sentence=sentence)
+        self.parser.update_data(data=data, fields=fields)
+
+        self.assertEqual(data["WIVWT"]["wind_angle_degree"], "0")
+        self.assertEqual(data["WIVWT"]["wind_angle_direction"], "L")
+        self.assertEqual(data["WIVWT"]["wind_speed_knots"], "1")
+        self.assertEqual(data["WIVWT"]["wind_speed_mps"], "2")
+        self.assertEqual(data["WIVWT"]["wind_speed_kph"], "3")
+
+    def test_update_sentence_gpgll(self):
```

Comment:
are these supposed to be testing anything?

Reply:
Those sentence readings are unimplemented.